feat: work with all yamllint config names

Zamiell opened the pull request:
Closes #22

Zamiell committed:
update

kcmr commented:
Hi @Zamiell,

Thank you for using the extension and for your contribution!
The behavior you described is not a bug, it was intentionally implemented this way to prioritize the user’s config value. That said, I’m fine with adding this feature and extending the search to all supported file types.

Before I can approve your PR, could you please update the README to reflect this new behavior?

Thanks again for your work on this. 🙌

kcmr reviewed (changes requested):
Please consider Copilot’s suggestion. It would be great if you could update the README so it reflects the new behavior. Thanks!

Zamiell committed:
update

Zamiell commented:
Hi kcmr,

Thanks for the quick review! I have spent some more time on this pull request and have made some more changes, which I think significantly improve the codebase.

Specifically:

- Fleshed out the README.md, as you requested.
- Removed YamlLintFixConfig:
  - First, having it exported at all was a bug, as it was not imported anywhere in the codebase. I highly recommend using `knip` to find bugs like these.
  - Second, typing it manually like that is not as good as just using TypeScript's inference, because it can get out of date, like exactly what happened in this pull request.
  - Thus, I removed it and added run-time assertions, which is considered best practice.
- I added some of the logic that I originally didn't bother with in order to make things more robust to how yamllint works. (YAMLLINT_CONFIG_FILE, XDG_CONFIG_HOME, home dir)
- I added strict to the tsconfig, because it was necessary for the `unknown` narrowing to work properly. Is there a reason that it wasn't there already? Virtually all TypeScript projects have this turned to true and it is considered best practice. As an aside, I would actually recommend going one step further and using `@tsconfig/strictest`, which I use in all my projects.

Zamiell commented:
@kcmr Did you get a chance to take a look at this PR?

kcmr commented:
@Zamiell No. Feel free to fork the repo, start from scratch or whatever you need and publish on the store.
